lta/rest_server.py

```diff
@@ -427,20 +490,35 @@ class TransferRequestActionsPopHandler(BaseLTAHandler):
         except Exception:
             raise tornado.web.HTTPError(400, reason="limit is not an int")
         pop_body = json_decode(self.request.body)
+        # find unclaimed or old transfer requests for the specified source
+        sdtr = self.db.TransferRequests
+        old_age = self.check_claims.old_age()
+        query = {
+            "$and": [
+                {"source": {"$regex": f"^{src}"}},
+                {"$or": [
+                    {"claimed": False},
+                    {"claim_time": {"$lt": f"{old_age}"}}
+                ]}
+            ]
+        }
         ret = []
-        for req in self.db['TransferRequests'].values():
-            if (req['source'].split(':', 1)[0] == src and
-                    (req['claimed'] is False or
-                     self.check_claims.old_claim(req['claim_time']))):
-                ret.append(req)
-                req['claimant'] = pop_body
-                req['claimed'] = True
-                req['claim_time'] = now()
-                limit -= 1
-                if limit <= 0:
-                    break
+        async for row in sdtr.find(query, REMOVE_ID, limit=limit):
+            row["claimant"] = pop_body
+            row["claimed"] = True
+            row["claim_time"] = now()
+            update_query = {"uuid": row["uuid"]}
+            update_doc = {"$set": row}
+            ret2 = await sdtr.find_one_and_update(update_query,
+                                                  update_doc,
+                                                  REMOVE_ID,
+                                                  return_document=AFTER)
+            if not ret2:
+                raise tornado.web.HTTPError(500, reason="unable to update transfer request")
```

Comment:
Similar to above pop function, needs to work properly in parallel instead of raising an error.